src/main/scala/com/mwz/sonar/scala/sensor/Measures.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Sonar Scala Plugin
+ * Copyright (C) 2018 All contributors
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package com.mwz.sonar.scala.sensor
+
+import scala.annotation.tailrec
+import scalariform.lexer.{Token, Tokens}
+
+/**
+ * Scala Sensor Metrics.
+ *
+ * @author mwz
+ * @author BalmungSan
+ */
+object Measures {
```

Comment:
I also change the **countClasses** and **countMethods** _methods_ from the imperative implementation to a functional one.